src/Bootstrap/Base.php

```diff
@@ -139,70 +145,16 @@ protected function _initLogging()
         } else {
             $logFilename = 'opus.log';
             if (! array_key_exists('SERVER_PROTOCOL', $_SERVER)
-                and ! array_key_exists('REQUEST_METHOD', $_SERVER)) {
+               and ! array_key_exists('REQUEST_METHOD', $_SERVER)) {
                 $logFilename = "opus-console.log";
             }
         }
 
-        $logfilePath = $config->workspacePath . '/log/' . $logFilename;
-
-        $logfile = @fopen($logfilePath, 'a', false);
-
-        if ($logfile === false) {
-            $path = dirname($logfilePath);
-
-            if (! is_dir($path)) {
-                throw new Exception('Directory for logging does not exist');
-            } else {
-                throw new Exception('Failed to open logging file:' . $logfilePath);
-            }
-        }
-
-        $GLOBALS['id_string'] = uniqid(); // Write ID string to global variables, so we can identify/match individual runs.
-
-        $format = '%timestamp% %priorityName% (%priority%, ID '.$GLOBALS['id_string'].'): %message%' . PHP_EOL;
-        $formatter = new \Zend_Log_Formatter_Simple($format);
-
-        $writer = new \Zend_Log_Writer_Stream($logfile);
-        $writer->setFormatter($formatter);
-
-        $logger = new \Zend_Log($writer);
-        $logLevelName = 'INFO';
-        $logLevelNotConfigured = false;
-
-        if (isset($config->log->level)) {
-            $logLevelName = strtoupper($config->log->level);
-        } else {
-            $logLevelNotConfigured = true;
-        }
-
-        $zendLogRefl = new \ReflectionClass('Zend_Log');
-
-        $invalidLogLevel = false;
-
-        $logLevel = $zendLogRefl->getConstant($logLevelName);
-
-        if (empty($logLevel)) {
-            $logLevel = Zend_Log::INFO;
-            $invalidLogLevel = true;
-        }
-
-        // filter log output
-        $priorityFilter = new \Zend_Log_Filter_Priority($logLevel);
-        \Zend_Registry::set('LOG_LEVEL', $logLevel);
-        $logger->addFilter($priorityFilter);
-
-        if ($logLevelNotConfigured) {
-            $logger->warn('Log level not configured, using default \'' . $logLevelName . '\'.');
-        }
-
-        if ($invalidLogLevel) {
-            $logger->err('Invalid log level \'' . $logLevelName .
-                    '\' configured.');
-        }
+        $logService = LogService::getInstance();
 
+        $logService->createLog('default', $logFilename);
+        $logger = $logService->addLog();
```

Comment:
I kept the `addLog()` function seperate from `createLog()` so that external modules can pass custom logs to `addLog()`. It is something that you mentioned in the comments of my previous commit.

Reply:
The function createLog and addLog should both exit. However they shouldn't work the way they do right now and addLog should not be used here.

Reply:
Oh, and I make mistakes, so if you see problems with my suggestions feel free to argue. Nevertheless `addLog` does not behave like I would expect it to and I think this will become more clear once the API of `LogService` is documented in the function and class comments.